pkgs/tools/networking/mosh/default.nix

```diff
@@ -10,15 +11,21 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ autoreconfHook pkgconfig ];
-  buildInputs = [ protobuf ncurses zlib IOTty makeWrapper perl openssl bash-completion ];
+  buildInputs = [ protobuf ncurses zlib IOTty makeWrapper perl openssl bash-completion ] ++ lib.optional withUtempter libutempter;
 
-  patches = [ ./ssh_path.patch ];
+  patches = [ ./ssh_path.patch ./utempter_path.patch ];
   postPatch = ''
     substituteInPlace scripts/mosh.pl \
         --subst-var-by ssh "${openssh}/bin/ssh"
   '';
 
-  configureFlags = [ "--enable-completion" ];
+  postFixup = lib.optionalString withUtempter ''
+    wrapProgram $out/bin/mosh-server \
+      --prefix PATH : "/run/wrappers/bin" \
+      --suffix PATH : "${libutempter}/lib/utempter"
+  '';
```

Comment:
This `wrapProgram` doesn't make much sense to me, I think it can be deleted: 
On NixOS systems, `/run/wrappers/bin` is already in `PATH`.
On non-NixOS systems, putting `${libutempter}/bin/utempter` in `PATH` is useless because it's in the nix store and lacks the privileges (setgid). In this case the user must manually copy the helper program to a location in PATH but outside the nix store, and set ownership and permissions accordingly. We cannot automate this.